packages/jwt-middleware/index.js

```diff
@@ -0,0 +1,88 @@
+'use strict';
+
+const Assert = require('assert');
+const ExpressJwt = require('express-jwt');
+const jws = require('jws');
+const JwksRsa = require('jwks-rsa');
+
+const WT_AUTH_EXEC = 'wt-authorize-execution';
+
+module.exports = () => {
+
+    return function(req, res, next) {
+
+        const ctx = req.webtaskContext;
+    
+        const iss = ctx.meta['iss'];
+        const aud = ctx.meta['aud'];
+
+        // if wt-authorize-execution is set to true (non-zero) then proceed with authn and authz
+
+        if (ctx.meta[WT_AUTH_EXEC] && ctx.meta[WT_AUTH_EXEC] !== "0"){
+
+            if (req.headers['authorization']) {
+
+                const match = (req.headers['authorization'] || '')
+                .trim()
+                .match(/^bearer (.+)$/i);
+
+                let jwt;
+
+                try {
+                    jwt = jws.decode(match[1]);
+                    if (!jwt) throw new Error();
+                } catch(e){
+                    const error = new Error('Invalid authorization header');
+                    error.statusCode = 401;
+                    return next(error);
+                }
+
+                // verify claims
+
+                if (
+                    jwt.payload.scope.indexOf("wt:admin") === -1 
+                    && jwt.payload.scope.indexOf(`wt:owner:${req.x_wt.container}`) === -1
+                    && jwt.payload.scope.indexOf(ctx.meta['wt-execution-scope']) === -1
+                ) {
+                    const error = new Error('Missing required scopes');
+                    error.statusCode = 401;
+                    return next(error);
+                }
+
+            } else {
+                const error = new Error('Missing authorization header');
+                error.statusCode = 401;
+                return next(error);
+            }
+
+            // verify if token is correctly signed and has correct aud/iss fields
+            
+            try {
+
+                const loadRsaKey = JwksRsa.expressJwtSecret({
+                    cache: true,
+                    rateLimit: true,
+                    jwksRequestsPerMinute: 5,
+                    jwksUri: `${iss}`
```

Comment:
I am not sure if you need to append the `/.well-known/jwks.json` to the `iss` before passing it to expressJwtSecret, or whether it is already done inside.

Reply:
@ggoodman stated that we cannot assume that `/.well-known/jwks.json` will be on the uri

Reply:
Well this is what OpenID connect discovery spec requires, so in all practicality it will be there. Runtime currently makes this assumption. But we can revisit. @ggoodman thoughts?

Reply:
@tjanczuk To me this falls in the realm of intelligent defaults. What if we were using the mock issuers we put together in our recent workshop? Since these don't support routing to `${iss}/.well-known/jwks.json`, we would be stuck.

On the other hand, our clients can make these sort of inferences by checking the `.well-known/jwks.json` suffix and falling back to whatever is supplied otherwise.

Reply:
Can we make the path suffix an optional metadata property on the webtask?

Reply:
@ggoodman Actually the mock issuers we used during the exercise did not care at all about the request path, so they worked well with this approach. But I am OK with being explicit if we specify this.